Modules/Core/Common/test/itkNeighborhoodOperatorTest.cxx

```diff
@@ -16,14 +16,52 @@
  *
  *=========================================================================*/
 
+#include "itkAnnulusOperator.h"
+#include "itkBackwardDifferenceOperator.h"
 #include "itkDerivativeOperator.h"
 #include "itkForwardDifferenceOperator.h"
-#include "itkBackwardDifferenceOperator.h"
+#include "itkGaussianDerivativeOperator.h"
 #include "itkGaussianOperator.h"
+#include "itkImageKernelOperator.h"
 #include "itkLaplacianOperator.h"
+#include "itkNeighborhoodOperator.h"
 #include "itkSobelOperator.h"
+
 namespace
 {
+template <typename T>
+constexpr bool
+IsDefaultConstructibleCopyableNoThrowMovableAndDestructible()
+{
+  return std::is_default_constructible<T>::value && std::is_copy_constructible<T>::value &&
+         std::is_copy_assignable<T>::value && std::is_nothrow_move_constructible<T>::value &&
+         std::is_nothrow_move_assignable<T>::value && std::is_nothrow_destructible<T>::value;
+}
+
+static_assert(std::is_copy_assignable<itk::NeighborhoodOperator<int, 3>>::value,
+              "NeighborhoodOperator be copy-assignable.");
+static_assert(std::is_nothrow_move_assignable<itk::NeighborhoodOperator<int, 3>>::value,
+              "NeighborhoodOperator be noexcept move-assignable.");
+
+static_assert(IsDefaultConstructibleCopyableNoThrowMovableAndDestructible<itk::AnnulusOperator<int>>(),
+              "AnnulusOperator should be default-constructible, copyable, noexcept movable, and destructible.");
+static_assert(IsDefaultConstructibleCopyableNoThrowMovableAndDestructible<itk::DerivativeOperator<int>>(),
+              "DerivativeOperator should be default-constructible, copyable, noexcept movable, and destructible.");
+static_assert(
```

Comment:
`BackwardDifferenceOperator` is missing from this list of asserts

Reply:
> `BackwardDifferenceOperator` is missing from this list of asserts

Very well observed, @dzenanz ! I skipped that one, because it does `ITK_DISALLOW_COPY_AND_ASSIGN(BackwardDifferenceOperator)`! https://github.com/InsightSoftwareConsortium/ITK/blob/v5.1b01/Modules/Core/Common/include/itkBackwardDifferenceOperator.h#L48

The "Rule of Zero" would allow copy-construction and assignment, for `BackwardDifferenceOperator`. It seems like that was not intended. I don't know why specifically `BackwardDifferenceOperator` does not allow copying.

The other eight `NeighborhoodOperator` derived classes already allowed copying before, and they still do with this PR. (Only with less code, and added nothrow-movability).

Reply:
You could add a new commit, to this PR or a follow-up PR, which removes `ITK_DISALLOW_COPY_AND_ASSIGN` from _BackwardDifferenceOperator_, in order to make things uniform. @hjmjohnson and @blowekamp what do you think?

Reply:
I leave it up to you @dzenanz @hjmjohnson @blowekamp whether or not you want this PR to remove `ITK_DISALLOW_COPY_AND_ASSIGN(BackwardDifferenceOperator)`. Please let me know 😃

Reply:
Let's merge this now, and fix `BackwardDifferenceOperator` in another PR.